pkg/resource/deploy/step_generator.go

```diff
@@ -1608,29 +1608,15 @@ func issueCheckFailures(printf func(*diag.Diag, ...interface{}), new *resource.S
 func processIgnoreChanges(inputs, oldInputs resource.PropertyMap,
 	ignoreChanges []string,
 ) (resource.PropertyMap, result.Result) {
-	ignoredInputs := resource.NewObjectProperty(inputs.Copy())
+	ignoredInputs := inputs.Copy()
```

Comment:
If you're rewriting this code, any thoughts on how to make it better?

I don't like that there are correctness assumptions that the caller will do

```go
input = processIgnoreChanges(inputs, oldInputs, resource.PropertyMap)
```

where we have assumptions on the future usage by the caller on diff, oldInputs, inputs.

Edit: I have a suggestion in a later comment.